### PR TITLE
Search improvements & bug fixes

### PR DESCRIPTION
**Pull request by max-zilla**
THIS PR CHANGES THE DATASET MODEL from author -> creator, if you see weird errors try deleting your volumes for mongo + elasticsearch and starting with clean db.

Added a ReactiveBase import to the search page that was causing rendering errors. As it is I can use basic and advanced search:
image

image

### CODE REVIEW

**Comment by longshuicy** [comment]
weird after i drop the volumn i see this 
image

I tried to create and add user metadata but it doesn't help. Can anyone else try? Wondering if it's just me

**Comment by tcnichol** [comment]
@longshuicy what did you put in the search box? i'm trying to replicate the error.

File: backend/app/routers/elasticsearch.py [hunk 1]
```diff
@@ -36,4 +36,8 @@ async def search_metadata(request: Request):
 async def search_file_dataset_and_metadata(request: Request):
     es = await connect_elasticsearch()
     query = await request.body()
-    return search_index(es, ["file", "dataset", "metadata"], query)
+    print("_MSEARCH")
+    print(query)
+    r = search_index(es, ["file", "dataset", "metadata"], query)
+    print(r)
```
Comment (longshuicy) [inline backend/app/routers/elasticsearch.py:42]: are those prints needed?

**Review by longshuicy — COMMENTED** [review]
Will there be any downstream impact of changing "author" to "creator", and "download" to "downloads"? I.e. frontend?

File: backend/app/routers/elasticsearch.py (hunk 1 above)
Comment (max-zilla) [reply backend/app/routers/elasticsearch.py]: no, removed

**Review by longshuicy — APPROVED** [review]

**Review by lmarini — CHANGES_REQUESTED** [review]
Found one error with renaming datasets.

File: backend/app/models/metadata.py [hunk 2]
```diff
@@ -329,9 +320,6 @@ async def patch_metadata(
         db["metadata"].replace_one(
             {"_id": metadata["_id"]}, MetadataDB(**metadata).to_mongo()
         )
-        # Update entry to the metadata index
-        doc = {"doc": {"content": metadata["content"]}}
-        update_record(es, "metadata", doc, metadata["_id"])
```
[the remark this answers is not in the document]
Comment (max-zilla) [reply backend/app/models/metadata.py:334]: resolved this

**Review by lmarini — APPROVED** [review]

**Pull request merged**